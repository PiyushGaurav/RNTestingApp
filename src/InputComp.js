import React, {useState} from 'react';
import {StyleSheet, Text, View, TextInput, Button} from 'react-native';

const InputComp = () => {
  const [name, setName] = useState('');

  function onSubmit() {
    let newName = name.trim();
    setName(newName);
  }

  return (
    <View style={styles.container}>
      <TextInput
        style={styles.input}
        value={name}
        placeholder={'enter your name'}
        onChangeText={name => setName(name)}
      />
      <Button
        testID="submit"
        style={styles.btn}
        onPress={onSubmit} // This line
        title="Submit"
      />
    </View>
  );
};

export default InputComp;

const styles = StyleSheet.create({
  container: {
    alignItems: 'center',
    padding: 24,
    borderBottomLeftRadius: 10,
    borderBottomRightRadius: 10,
  },
  input: {
    fontWeight: 'bold',
    color: 'white',
    borderWidth: 1,
    borderColor: '#333',
    width: '90%',
    padding: 10,
    marginVertical: 15,
    borderRadius: 5,
    color: 'black',
  },
  btn: {
    height: 40,
    width: '70%',
    backgroundColor: '#333',
    justifyContent: 'center',
    alignItems: 'center',
    borderRadius: 20,
    shadowColor: '#000',
    shadowOffset: {
      width: 0,
      height: 6,
    },
    shadowOpacity: 0.39,
    shadowRadius: 8.3,
    elevation: 13,
  },
  btnText: {
    color: 'white',
  },
});
