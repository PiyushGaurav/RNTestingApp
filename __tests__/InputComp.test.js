import {render, fireEvent, screen} from '@testing-library/react-native';
import InputComp from '../src/InputComp';
import {create} from 'react-test-renderer';

describe('Input Comp', () => {
  test('renders correctly', () => {
    const tree = create(<InputComp />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders correctly', () => {
    render(<InputComp />);
    const input = screen.getByPlaceholderText('enter your name');

    // simulates the user focusing on TextInput and typing text one char at a time
    fireEvent.changeText(input, ' John Doe ');

    expect(input.props.value).toBe(' John Doe ');

    // simulates the user pressing on any pressable element
    const buttonElement = screen.getByTestId('submit');

    fireEvent.press(buttonElement);

    // expect(onSubmitMock).toHaveBeenCalledTimes(1);
    expect(input.props.value).toBe('John Doe');
  });
});
