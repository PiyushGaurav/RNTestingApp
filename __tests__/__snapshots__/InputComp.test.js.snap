// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Comp renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "borderBottomLeftRadius": 10,
      "borderBottomRightRadius": 10,
      "padding": 24,
    }
  }
>
  <TextInput
    onChangeText={[Function]}
    placeholder="enter your name"
    style={
      {
        "borderColor": "#333",
        "borderRadius": 5,
        "borderWidth": 1,
        "color": "black",
        "fontWeight": "bold",
        "marginVertical": 15,
        "padding": 10,
        "width": "90%",
      }
    }
    value=""
  />
  <View
    accessibilityRole="button"
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
    testID="submit"
  >
    <View
      style={
        [
          {},
        ]
      }
    >
      <Text
        style={
          [
            {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Submit
      </Text>
    </View>
  </View>
</View>
`;
